### ISSUE TITLE
[REVIEW]: DrWatson: the perfect sidekick for your scientific inquiries

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2673]
**Submitting author:** @tamasgal (<a href="http://orcid.org/0000-0001-7821-8673">Tamás Gál</a>)
**Repository:** <a href="https://github.com/JuliaDynamics/DrWatson.jl" target ="_blank">https://github.com/JuliaDynamics/DrWatson.jl</a>
**Version:** 1.16.3
**Editor:** @dpsanders
**Reviewers:** @jpfairbanks, @kescobo, @apdavison
**Archive:** <a href="https://doi.org/10.5281/zenodo.4133570" target="_blank">10.5281/zenodo.4133570</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/58375662785c6e80e2849fc2165471aa/status.svg)](https://joss.theoj.org/papers/58375662785c6e80e2849fc2165471aa)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/58375662785c6e80e2849fc2165471aa"><img src="https://joss.theoj.org/papers/58375662785c6e80e2849fc2165471aa/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/58375662785c6e80e2849fc2165471aa/status.svg)](https://joss.theoj.org/papers/58375662785c6e80e2849fc2165471aa)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/JuliaDynamics/DrWatson.jl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@jpfairbanks & @kescobo, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @dpsanders know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JuliaDynamics/DrWatson.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@tamasgal) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JuliaDynamics/DrWatson.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@tamasgal) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

---

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JuliaDynamics/DrWatson.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@tamasgal) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @jpfairbanks, @kescobo it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by dpsanders** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #2673 with the following error: 

 sh: 0: getcwd() failed: No such file or directory
pandoc: 10.21105.joss.02673.pdf: openBinaryFile: does not exist (No such file or directory)
Looks like we failed to compile the PDF

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.jsc.2003.04.002 is OK
- doi:10.1201/9781315373461-3 is OK
- doi:10.5063/F1GF0RF6 is OK
- 10.6084/m9.figshare.3115156.v2 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02673/joss.02673/10.21105.joss.02673.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02673/joss.02673/10.21105.joss.02673.pdf) :page_facing_up: :point_left:

**Comment by dpsanders** [comment]
👋 Hi @jpfairbanks and @kescobo. Thank you for agreeing to review this submission. Please work your way through the checklist above of items. Please feel free to raise technical issues on the repository, and procedural questions here. Thanks!

**Comment by dpsanders** [comment]
I would say no, that's up to the authors themselves.

**Comment by kescobo** [comment]
I've completed my review - pending the changes [requested here](https://github.com/JuliaDynamics/DrWatson.jl/pull/192), I'm happy to sign off.

**Comment by jpfairbanks** [comment]
So I have a question about scope. Is the JOSS paper format intended to be a complete description of the code? How much of the "explaining how to use the software" should be delegated to the docs (which will stay up to date).

**Comment by dpsanders** [comment]
Thanks @kescobo!

@jpfairbanks: That's a great question.
The JOSS paper is definitely *not* a complete description of the code.

There should be one or a few code snippets giving a feeling for how to use the code, to explain the goals etc. of the software. 

The docs should have a complete API description etc.

**Comment by dpsanders** [comment]
@jpfairbanks You can find review criteria here:

https://joss.readthedocs.io/en/latest/review_criteria.html

**Comment by dpsanders** [comment]
The idea is for the paper to be relatively short, with the main documentation in the software repository itself.

**Comment by jpfairbanks** [comment]
Thanks, I had read some of that language above in this thread and got confused because it talks about API docs, but I’m supposed to check that the API is documented in the actual manual. Not that the paper contains docs. This now makes sense. BTW I can’t mark the checkboxes in the checklist. Am I supposed to have different repo privs or something?

**Comment by dpsanders** [comment]
OK, let me know if the wording can be improved.

I believe that you should have got an invitation to the JOSS account giving you those privileges?

**Comment by Datseris** [comment]
> Does JOSS have the concept of "corresponding author" and/or author order conventions?

For this paper you can consider myself the corresponding author (if need be).

**Comment by jpfairbanks** [comment]
I think that this is a great tool and a great write-up that sufficiently introduces the software, situates it in the relevant prior art and motivates me to adopt the software based on claimed benefits. The benefits of better simulation data management is obvious to anyone who has analyzed the performance, accuracy, or predictions of a computer simulation. There is no empirical evidence of efficiency or utility presented, but I am not sure that publication of this material requires that kind of empirical validation and defer to the editor's judgement.

The biggest issue from a methodological perspective is that Dr. Watson avoids the use of databases for managing the data collected during the experiments. The current practice for simulation data management is to create files with paths that encode information about the parameters and with contents that encode information about the results. Dr. Watson automates and formalizes these ad-hoc techniques. In that automating and formalizing ad hoc techniques can improve the reliability of the methods, this is good. But as a field, computer science has solved this data management problem with an entirely different technique.

Relational Database Management Systems (RDBMS) are the standard technology designed for managing data when you need persistence and correctness with the ability to store data quickly and query against it later. RDBMSes support flexible query languages for answering complex, ad-hoc queries about the data and indexing to make it fast. Why does Dr. Watson not use a database for storing this data? The paper should address this approach and why it was not selected. The current approach for simulation data management is ad hoc file naming conventions and folder structures containing structured or semi-structured data in the output files. So any solution that starts by improving this solution is welcome. However for more fruitful analysis of the resulting data, a proper data ingestion with ETL into an SQL database would allow for more rich querying. This could lead to easier feature development down the road when it comes to experiment analysis.

This concern could be addressed by identifying previous attempts to use RDBMSes in simulation data management and explaining why they failed, or discussing this path as a viable approach that has not yet been explored, and leaving integration with existing RDBMS techniques for future work, once the RDBMS ecosystem in Julia has matured.

**Comment by Datseris** [comment]
Thank you for the review @jpfairbanks , [CaosDB](https://caosdb.org/) is a RDBMS (I believe, although the creators name it Scientific Data Management System), and has been developed by scientists for scientists. We plan to integrate DrWatson with CaosDB.jl once a sufficiently stable Julia implementation exists. I believe once such an integration is in order, your concerns about the lack of data managing capabilities of DrWatson will be addressed. However, I do think this will take substantial amount of effort and thus it is unlikely that we are able to do it during the period of this review.

**Comment by jpfairbanks** [comment]
That sounds great, then I would request that the article describe the limitation of DrWatson for organizing or querying the resulting data and that you would use DrWatson for tracking the data generation processes and then ETL it into CaosDB for querying. Describing that as future work would be appropriate.

**Comment by Datseris** [comment]
Fair point, addressed in https://github.com/JuliaDynamics/DrWatson.jl/commit/bbe9a83dffc0e13d96285c8713db93ac1b695e89 .

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02673/joss.02673/10.21105.joss.02673.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02673/joss.02673/10.21105.joss.02673.pdf) :page_facing_up: :point_left:

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02673/joss.02673/10.21105.joss.02673.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02673/joss.02673/10.21105.joss.02673.pdf) :page_facing_up: :point_left:

**Comment by Datseris** [comment]
@dpsanders we thank both reviewers for the constructive and swift review. In the latest version of the paper (generated above) we address all raised concerns.

**Comment by jpfairbanks** [comment]
I always forget that github invitations require interaction with email.

**Comment by jpfairbanks** [comment]
> Please note that DrWatson is not a data management system and provides only basic data management functionality that remains self-contained in a single scientific project. One of our main future goals is to integrate DrWatson with a Relational Data Management System, specifically CaosDB (Fitschen et al., 2019), which had been developed specifically to handle large data bases connecting several scientific projects.
👍

**Comment by dpsanders** [comment]
@whedon add @apdavison as reviewer

**Comment by whedon** [comment]
OK, @apdavison is now a reviewer

**Comment by dpsanders** [comment]
Thanks @apdavison for agreeing to review this as well. Dr. Andrew Davison is one of the main authors of the Python `sumatra` package.

**Comment by Datseris** [comment]
Hello, may we please ask for an update here? @apdavison

**Comment by apdavison** [comment]
My apologies for the delay, I'll try to finish the review this week.

**Comment by apdavison** [comment]
@dpsanders I can't edit the checklist. When I click to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations it says the invitation has expired.

**Comment by dpsanders** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the sole reviewer of this submission
@whedon assign @username as reviewer

# Add a GitHub user to the reviewers of this submission
@whedon add @username as reviewer

# Re-invite a reviewer (if they can't update checklists)
@whedon re-invite @username as reviewer

# Remove a GitHub user from the reviewers of this submission
@whedon remove @username as reviewer

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Set the software version at the top of the issue e.g.
@whedon set v1.0.1 as version

# Open the review issue
@whedon start review

EDITORIAL TASKS

# All commands can be run on a non-default branch, to do this pass a custom 
# branch name by following the command with `from branch custom-branch-name`.
# For example:

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Remind an author or reviewer to return to a review after a
# certain period of time (supported units days and weeks)
@whedon remind @reviewer in 2 weeks

# Ask Whedon to do a dry run of accepting the paper and depositing with Crossref
@whedon accept

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

EiC TASKS

# Invite an editor to edit a submission (sending them an email)
@whedon invite @editor as editor

# Reject a paper
@whedon reject

# Withdraw a paper
@whedon withdraw

# Ask Whedon to actually accept the paper and deposit with Crossref
@whedon accept deposit=true

```

**Comment by dpsanders** [comment]
@whedon re-invite @apdavison as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@apdavison please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by dpsanders** [comment]
@apdavison I have re-invited you. Please let met know if it still doesn't work. Thanks!

**Comment by apdavison** [comment]
This seems like a nice toolkit for Julia users, which brings together a lot of functionality that would otherwise have to be assembled from several sources. I have the impression that a lot of the functionality relies on the modern design of Julia (in particular the package manager), and would be more difficult to implement in "messier" language ecosystems like Python's. The paper is well structured and clearly written.

I don't have any previous experience with Julia, so have not tried installing or using the package, but I think we can take the word of @kescobo and @jpfairbanks that this works!

My list of comments is not extensive, so I include it here rather than creating a new issue in the DrWatson repository.

- I don't think the "Community guidelines" item in the checklist has been well addressed. There is a CONTRIBUTING.md file, but as far as I can see the documentation itself doesn't contain "_guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support_".
- Tele'nczuk should be Teleńczuk.
- Sumatra is not just for Python projects. It is written in Python but has support for R, MATLAB, bash, and is designed to be easy to extend for other languages, and to be usable in heterogeneous environments and pipelines.
- I think [ReproZip](https://www.reprozip.org) and [ActivePapers](https://www.activepapers.org) should be mentioned in the "Comparison" section, as tools which also aim to make scientific projects easily portable, but in more language-independent ways.

**Comment by Datseris** [comment]
Thank you @apdavison for the quick response and your review. We have addressed your comments by:

1. Adding a "Support & Contributing" section in the docs.
1. Improving the introduction of sumatra
1. Mentioning ReproZip (but not ActivePapers because contrary to its name it doesn't feel active enough of a repository to us)
1. Addressing typos

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02673/joss.02673/10.21105.joss.02673.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02673/joss.02673/10.21105.joss.02673.pdf) :page_facing_up: :point_left:

**Comment by dpsanders** [comment]
@Datseris I agree with @apdavison that ActivePapers should be cited, even if it is not currently "active".

**Comment by Datseris** [comment]
Alright, we have added the citation to it in the latest version.

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02673/joss.02673/10.21105.joss.02673.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02673/joss.02673/10.21105.joss.02673.pdf) :page_facing_up: :point_left:

**Comment by dpsanders** [comment]
@Datseris @tamasgal I sent a few minor corrections to the paper.

Once that's done I think the paper is ready to accept!
Please archive it at Zenodo and let me know the version and DOI. You need to make sure that the title and author info in Zenodo match the paper exactly. Thanks!

**Comment by Datseris** [comment]
Hi @dpsanders thanks for the improvements to the paper.

Here is the Zenodo DOI: [10.5281/zenodo.4133570 ](https://zenodo.org/account/settings/github/repository/JuliaDynamics/DrWatson.jl#) 

> You need to make sure that the title and author info in Zenodo match the paper exactly.

The only problem is, I don't know how to do this... I didn't find a way in the website to alter the archived Zenodo information (and Zenodo archives GitHub repositories automatically). Do I have to make a manual Zenodo entry?

**Comment by kescobo** [comment]
@Datseris once you've linked the repo to Zenodo, every time you make a release (a github release, which may or may not happen automatically depending on your bots) it will automatically archive it

**Comment by Datseris** [comment]
Yes I have done this. But David has some additional requirements regarding author info and paper title _in Zenodo_. I do not see this info somewhere at the moment...

**Comment by kescobo** [comment]
Ahh, sorry I missed that. I think that you can manually edit the title and authors of your last release, and then it will automatically keep the same ones going forward. So:

1. Click "uploads" in the top banner

**Comment by Datseris** [comment]
Thanks so much @kescobo ! @dpsanders now everything is in order here: https://zenodo.org/record/4133570 !

**Comment by dpsanders** [comment]
I'm now wondering if the title should be `DrWatson.jl: the...` instead of just `DrWatson`?

**Comment by Datseris** [comment]
It is not clear to me whether I have to respond or do something :D

**Comment by kescobo** [comment]
> I'm now wondering if the title should be DrWatson.jl: the... instead of just DrWatson

`DrWatson` seems like the name of the software, `DrWatson.jl` is the package that implements it. I think either would work, but fine to leave as-is.

**Comment by dpsanders** [comment]
I guess my point is that it's not clear from the title that this is a Julia package. But I'm happy to go with it as is.

**Comment by dpsanders** [comment]
@whedon set version as 1.16.3

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by dpsanders** [comment]
@whedon set 1.16.3 as version

**Comment by whedon** [comment]
OK. 1.16.3 is the version.

**Comment by dpsanders** [comment]
@whedon set https://zenodo.org/record/4133570 as archive

**Comment by whedon** [comment]
https://zenodo.org/record/4133570 doesn't look like an archive DOI.

**Comment by dpsanders** [comment]
@Datseris Could you please check that the version number is correct

**Comment by dpsanders** [comment]
@whedon accept

**Comment by whedon** [comment]
No archive DOI set. Exiting...

**Comment by dpsanders** [comment]
@whedon set 10.5281/zenodo.4133570 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4133570" target="_blank">10.5281/zenodo.4133570</a> is the archive.

**Comment by dpsanders** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.jsc.2003.04.002 is OK
- doi:10.1201/9781315373461-3 is OK
- doi:10.5063/F1GF0RF6 is OK
- 10.6084/m9.figshare.3115156.v2 is OK
- 10.3390/data4020083 is OK
- 10.1145/2882903.2899401 is OK

MISSING DOIs

- None

INVALID DOIs

- https://doi.org/10.1016/j.procs.2011.04.061 is INVALID because of 'https://doi.org/' prefix
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1875

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1875, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@Datseris - could you please merge this PR which fixes some issues with your DOIs? https://github.com/JuliaDynamics/DrWatson.jl/pull/208

**Comment by Datseris** [comment]
Thank you @arfon , I just did!

**Comment by arfon** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.jsc.2003.04.002 is OK
- 10.1201/9781315373461-3 is OK
- 10.5063/F1GF0RF6 is OK
- 10.6084/m9.figshare.3115156.v2 is OK
- 10.3390/data4020083 is OK
- 10.1145/2882903.2899401 is OK
- 10.1016/j.procs.2011.04.061 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1879

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1879, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.jsc.2003.04.002 is OK
- 10.1201/9781315373461-3 is OK
- 10.5063/F1GF0RF6 is OK
- 10.6084/m9.figshare.3115156.v2 is OK
- 10.3390/data4020083 is OK
- 10.1145/2882903.2899401 is OK
- 10.1016/j.procs.2011.04.061 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1321736482017759232) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1880
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02673](https://doi.org/10.21105/joss.02673)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@jpfairbanks, @kescobo, @apdavison - many thanks for your reviews here and to @dpsanders for editing this submission ✨ 

@Datseris - your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02673/status.svg)](https://doi.org/10.21105/joss.02673)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02673">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02673/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02673/status.svg
   :target: https://doi.org/10.21105/joss.02673
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02673/status.svg)](https://doi.org/10.21105/joss.02673)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by dpsanders** [comment]
Congrats @tamasgal and @datseris!

Many thanks to @kescobo, @jpfairbanks and @apdavison for your reviews.
